File: javautilsconcurrent/src/main/java/com/hansen/javautilsconcurrent/MainActivity.java
package com.hansen.javautilsconcurrent;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.blankj.utilcode.util.LogUtils;
import com.hansen.utils.LogUtil;

import java.util.concurrent.BlockingQueue;

public class MainActivity extends AppCompatActivity {
    private String TAG = getPackageName().getClass().getName();
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


    }

    class Producer implements Runnable {
        private final BlockingQueue queue;
        Producer(BlockingQueue q) { queue = q; }
        @Override
        public void run() {
            try {
                while (true) { queue.put(produce()); }
            } catch (InterruptedException ex) { }
        }
        Object produce() {
            LogUtils.iTag(TAG,"produce");
            return "";
        };
    }

}

File: kotlindemo/src/main/java/com/hansen/kotlindemo/MainActivity.kt
package com.hansen.kotlindemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.hansen.kotlindemo.base.BaseActivity

class MainActivity : BaseActivity() {
    private val mTitles = arrayOf("主页","工作","设备","视频")

    /**
     * 未被选择的图标
     */
    private val mIconsUnSelecteds = intArrayOf()
    private val mIconsSelecteds = intArrayOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun start() {
    }

    override fun initView() {
    }

    override fun initData() {
    }

    override fun layoutId(): Int {
       return R.layout.activity_main
    }
}
